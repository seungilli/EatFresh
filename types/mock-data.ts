import { HistoryType, Meal } from "./data";

export const mockMeal1: Meal = {
    idMeal: "67890",
    strMeal: "Delicious Pasta",
    strMealThumb: "https://via.placeholder.com/150",
    strCategory: "Pasta",
    strArea: "Italian",
    strDrinkAlternate: null,
    strInstructions: "Boil pasta. Prepare sauce. Mix and serve.",
    strTags: "Quick,Easy",
    strYoutube: "https://youtube.com/test-video-pasta",
    strIngredient1: "Pasta",
    strIngredient2: "Tomato Sauce",
    strIngredient3: "Cheese",
    strIngredient4: "Garlic",
    strIngredient5: "Basil",
    strIngredient6: "Salt",
    strIngredient7: "Pepper",
    strIngredient8: "Olive Oil",
    strIngredient9: "Onion",
    strIngredient10: null,
    strIngredient11: null,
    strIngredient12: null,
    strIngredient13: null,
    strIngredient14: null,
    strIngredient15: null,
    strIngredient16: null,
    strIngredient17: null,
    strIngredient18: null,
    strIngredient19: null,
    strIngredient20: null,
    strMeasure1: "200 grams",
    strMeasure2: "1 cup",
    strMeasure3: "100 grams",
    strMeasure4: "2 cloves",
    strMeasure5: "5 leaves",
    strMeasure6: "to taste",
    strMeasure7: "to taste",
    strMeasure8: "2 tbsp",
    strMeasure9: "1 medium",
    strMeasure10: null,
    strMeasure11: null,
    strMeasure12: null,
    strMeasure13: null,
    strMeasure14: null,
    strMeasure15: null,
    strMeasure16: null,
    strMeasure17: null,
    strMeasure18: null,
    strMeasure19: null,
    strMeasure20: null,
    strSource: "Family Recipe",
    strImageSource: "https://via.placeholder.com/image-source1",
    strCreativeCommonsConfirmed: "Yes",
    dateModified: "2023-05-15",
};


export const mockMeal2: Meal = {
    idMeal: "54321",
    strMeal: "Spicy Chicken Curry",
    strMealThumb: "https://via.placeholder.com/200",
    strCategory: "Curry",
    strArea: "Indian",
    strDrinkAlternate: "Lassi",
    strInstructions: "Marinate chicken. Cook with spices. Serve hot.",
    strTags: "Spicy,Dinner",
    strYoutube: "https://youtube.com/test-video-curry",
    strIngredient1: "Chicken",
    strIngredient2: "Curry Powder",
    strIngredient3: "Yogurt",
    strIngredient4: "Garlic",
    strIngredient5: "Ginger",
    strIngredient6: "Onion",
    strIngredient7: "Tomato",
    strIngredient8: "Coriander",
    strIngredient9: "Chili Powder",
    strIngredient10: "Salt",
    strIngredient11: "Pepper",
    strIngredient12: "Oil",
    strIngredient13: "Cumin Seeds",
    strIngredient14: "Turmeric",
    strIngredient15: "Cardamom",
    strIngredient16: "Cinnamon Stick",
    strIngredient17: null,
    strIngredient18: null,
    strIngredient19: null,
    strIngredient20: null,
    strMeasure1: "500 grams",
    strMeasure2: "2 tbsp",
    strMeasure3: "1 cup",
    strMeasure4: "3 cloves",
    strMeasure5: "1 inch",
    strMeasure6: "2 medium",
    strMeasure7: "2 large",
    strMeasure8: "to garnish",
    strMeasure9: "1 tsp",
    strMeasure10: "to taste",
    strMeasure11: "to taste",
    strMeasure12: "3 tbsp",
    strMeasure13: "1 tsp",
    strMeasure14: "1/2 tsp",
    strMeasure15: "2 pods",
    strMeasure16: "1 stick",
    strMeasure17: null,
    strMeasure18: null,
    strMeasure19: null,
    strMeasure20: null,
    strSource: "Traditional Indian Cookbook",
    strImageSource: "https://via.placeholder.com/image-source2",
    strCreativeCommonsConfirmed: "No",
    dateModified: "2022-12-01",
};

export const mockHistoryItem1: HistoryType = {
    time: "2023-11-23T12:00:00Z",
    searchedText: "History Item 1",
};

export const mockHistoryItem2: HistoryType = {
    time: "2023-11-24T14:00:00Z",
    searchedText: "History Item 2",
};


