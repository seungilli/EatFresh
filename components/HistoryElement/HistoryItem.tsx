import { HistoryType } from "@/types/data";
import { View, StyleSheet } from "react-native";
import { Text } from "react-native-paper";

const styles = StyleSheet.create({
  historyItem: {
    flexDirection: "row",
    alignItems: "center",
    marginBottom: 10,
  },
  historyTime: {
    marginRight: 10,
  },
  historyText: {
    flex: 1,
    fontSize: 16,
    fontWeight: "bold",
  },
});

type HistoryItemProps = {
  item: HistoryType;
};

export default function HistoryItem({ item }: HistoryItemProps) {
  return (
    <View style={styles.historyItem}>
      <Text style={styles.historyTime}>{item.time}</Text>
      <Text style={styles.historyText}>{item.searchedText}</Text>
    </View>
  );
}
