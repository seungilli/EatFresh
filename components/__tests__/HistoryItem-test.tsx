import React from "react";
import renderer from "react-test-renderer";
import { HistoryType } from "@/types/data";
import HistoryItem from "../HistoryElement/HistoryItem";

describe("<HistoryItem />", () => {
  const mockItem: HistoryType = {
    time: "2024-11-24 12:00 PM",
    searchedText: "test1",
  };

  it("should render", () => {
    const tree = renderer.create(<HistoryItem item={mockItem} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
