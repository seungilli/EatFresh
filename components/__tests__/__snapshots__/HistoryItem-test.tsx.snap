// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HistoryItem Component renders correctly with given props 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flexDirection": "row",
      "marginBottom": 10,
    }
  }
>
  <Text
    style={
      [
        {
          "textAlign": "left",
        },
        {
          "color": "rgba(28, 27, 31, 1)",
          "fontFamily": "System",
          "fontWeight": "400",
          "letterSpacing": 0,
        },
        {
          "writingDirection": "ltr",
        },
        {
          "marginRight": 10,
        },
      ]
    }
  >
    2024-11-24 12:00 PM
  </Text>
  <Text
    style={
      [
        {
          "textAlign": "left",
        },
        {
          "color": "rgba(28, 27, 31, 1)",
          "fontFamily": "System",
          "fontWeight": "400",
          "letterSpacing": 0,
        },
        {
          "writingDirection": "ltr",
        },
        {
          "flex": 1,
          "fontSize": 16,
          "fontWeight": "bold",
        },
      ]
    }
  >
    test1
  </Text>
</View>
`;
